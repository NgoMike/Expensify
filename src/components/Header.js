import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Expensify</h1>

    <button>
      <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink> {/* NavLink activeClassName lets customize with css */}
    </button>

    <button>
      <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
    </button>

  </header>
);

export default Header;