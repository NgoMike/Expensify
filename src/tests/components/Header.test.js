import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';
// import toJSON from 'enzyme-to-json';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});

  // const renderer = new ReactShallowRenderer();
  // renderer.render(<Header />);
  // expect(renderer.getRenderOutput()).toMatchSnapshot();
  // expect(wrapper.find('h1').text()).toBe('Expensify');
  // expect(toJSON(wrapper)).toMatchSnapshot();