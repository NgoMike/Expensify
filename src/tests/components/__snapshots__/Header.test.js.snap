// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <button>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
     
  </button>
  <button>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      to="/create"
    >
      Create Expense
    </NavLink>
  </button>
</header>
`;
